Add qa model and new settings in ml-commons (#6749)
opensearch-project/documentation-website: _ml-commons-plugin/custom-local-models.md

As of OpenSearch 2.12, OpenSearch supports local cross-encoder models.

As of OpenSearch 2.13, OpenSearch supports local question answering models.

Running local models on the CentOS 7 operating system is not supported. Moreover, not all local models can run on all hardware and operating systems.
{: .important}

## Preparing a model

> For all the models, you must provide a tokenizer JSON file within the model zip file.

Reviewer: ZIP file?

Reviewer 2: I don't think zip file should be capitalized.

## Step 5: Use the model for search

To learn how to use the model for vector search, see [Using an ML model for neural search]({{site.url}}{{site.baseurl}}/search-plugins/neural-search/#using-an-ml-model-for-neural-search).

> ## Question Answering models

Reviewer: ```suggestion
## Question answering models
```

> Question Answering model extracts the answer to the question from a given context. Currently ml-commons supports context in `text` format.

Reviewer: ```suggestion
A question answering model extracts the answer to a question from a given context. ML Commons supports context in `text` format.
```

To register a question answering model, send a request in the following format. Specify the `function_name` as `QUESTION_ANSWERING`:

```json
POST /_plugins/_ml/models/_register
{
    "name": "question_answering",
    "version": "1.0.0",
    "function_name": "QUESTION_ANSWERING",
    "description": "test model",
    "model_format": "TORCH_SCRIPT",
    "model_group_id": "lN4AP40BKolAMNtR4KJ5",
    "model_content_hash_value": "e837c8fc05fd58a6e2e8383b319257f9c3859dfb3edc89b26badfaf8a4405ff6",
    "model_config": { 
        "model_type": "bert",
        "framework_type": "huggingface_transformers"
    },
    "url": "https://github.com/opensearch-project/ml-commons/blob/main/ml-algorithms/src/test/resources/org/opensearch/ml/engine/algorithms/question_answering/question_answering_pt.zip?raw=true"
}
```
{% include copy-curl.html %}

Then send a request to deploy the model:

```json
POST _plugins/_ml/models/<model_id>/_deploy
```
{% include copy-curl.html %}

> To test a question answering, send the following request. It requires a `question` and the relevant `context` from which answer will be generated:

Reviewer: ```suggestion
To test a question answering model, send the following request. It requires a `question` and the relevant `context` from which the answer will be generated:
```

```json
POST /_plugins/_ml/_predict/question_answering/<model_id>
{
  "question": "Where do I live?"
  "context": "My name is John. I live in New York"
}
```
{% include copy-curl.html %}

> The response gives the answer based on the context provided:

Reviewer: ```suggestion
The response provides the answer based on the context:
```
